components/nexus-httpclient/src/main/java/org/sonatype/nexus/httpclient/config/ConfigurationCustomizer.java

```diff
@@ -215,86 +199,17 @@ NexusHttpRoutePlanner createRoutePlanner(final ProxyConfiguration proxy) {
     if (proxy.getNonProxyHosts() != null) {
       patterns.addAll(Arrays.asList(proxy.getNonProxyHosts()));
     }
-    patterns.addAll(DEFAULT_NO_PROXY_HOSTS_PATTERN_STRINGS);
     String nonProxyPatternString = Joiner.on("|").join(Iterables.transform(patterns, GLOB_STRING_TO_REGEXP_STRING));
-    Pattern nonProxyPattern;
-    try {
-      nonProxyPattern = Pattern.compile(nonProxyPatternString, Pattern.CASE_INSENSITIVE);
-    }
-    catch (PatternSyntaxException e) {
-      log.warn("Invalid non-proxy host regex: {}, using defaults", nonProxyPatternString, e);
-      nonProxyPattern = DEFAULT_NO_PROXY_HOSTS_PATTERN;
-    }
-    syncHttpSystemProperties(proxy);
-    return new NexusHttpRoutePlanner(proxies, nonProxyPattern);
-  }
-
-  private void syncHttpSystemProperties(final ProxyConfiguration proxy) {
-    // HTTP proxy
-    ProxyServerConfiguration http = proxy.getHttp();
-    if (http != null && http.isEnabled()) {
-      System.setProperty("http.proxyHost", http.getHost());
-      System.setProperty("http.proxyPort", Integer.toString(http.getPort()));
-      if (http.getAuthentication() != null) {
-        if (http.getAuthentication() instanceof UsernameAuthenticationConfiguration) {
-          UsernameAuthenticationConfiguration usernamePassword = (UsernameAuthenticationConfiguration) http
-              .getAuthentication();
-          System.setProperty("http.proxyUser", usernamePassword.getUsername());
-          System.setProperty("http.proxyPassword", usernamePassword.getPassword());
-        }
-        else {
-          log.warn("Authentication {} not supported for Java Networking, system properties not set",
-              http.getAuthentication().getClass().getSimpleName());
-        }
+    Pattern nonProxyPattern = null;
+    if (!Strings2.isBlank(nonProxyPatternString)) {
+      try {
+        nonProxyPattern = Pattern.compile(nonProxyPatternString, Pattern.CASE_INSENSITIVE);
       }
-      else {
-        System.clearProperty("http.proxyUser");
-        System.clearProperty("http.proxyPassword");
+      catch (PatternSyntaxException e) {
+        log.warn("Invalid non-proxy host regex: {}, using defaults", nonProxyPatternString, e);
```

Comment:
We at NX never emitted ERROR logs unless they made the system go down too. Hence, in this case, when nonProxyHosts detected as invalid, but everything goes along (system continues with empty nonProxyHosts), WARN is completely fine IMHO.

OTOH, given the validation now in place, this catch may never trigger, as this is leftover from pre-validator times.